sys/include/suit/v3/suit.h

```diff
@@ -0,0 +1,225 @@
+};
+
+/**
+ * @brief SUIT v3 component struct
+ */
+typedef struct {
+    uint32_t size;                      /**< Size */
+    nanocbor_value_t identifier;        /**< Identifier*/
+    nanocbor_value_t url;               /**< Url */
+    nanocbor_value_t digest;            /**< Digest */
+} suit_v3_component_t;
+
+/**
+ * @brief SUIT manifest struct
+ */
+typedef struct {
+    const uint8_t *buf;             /**< ptr to the buffer of the manifest */
+    size_t len;                     /**< length of the manifest */
+    const uint8_t *cose_payload;    /**< ptr to the payload of the COSE sign */
+    size_t cose_payload_len;        /**< length of the COSE payload */
+    uint32_t validated;             /**< bitfield of validated policies */
+    uint32_t state;                 /**< bitfield holding state information */
+
+    /** List of components in the manifest */
+    suit_v3_component_t components[SUIT_V3_COMPONENT_MAX];
+    unsigned components_len;        /**< Current number of components */
+    uint32_t component_current;     /**< Current component index */
+    riotboot_flashwrite_t *writer;  /**< Pointer to the riotboot flash writer */
+    /** Manifest validation buffer */
+    uint8_t validation_buf[SUIT_COSE_BUF_SIZE];
+    char *urlbuf;                   /**< Buffer containing the manifest url */
+    size_t urlbuf_len;              /**< Length of the manifest url */
+} suit_v3_manifest_t;
+
+/**
+ * @brief Bit flags used to determine if SUIT manifest contains components
+ */
+#define SUIT_MANIFEST_HAVE_COMPONENTS   (0x1)
+/**
+ * @brief Bit flags used to determine if SUIT manifest contains an image
+ */
+#define SUIT_MANIFEST_HAVE_IMAGE        (0x2)
+
+/**
+ * @brief Parse a manifest
+ *
+ * @note The buffer is still required after parsing, please don't reuse the
+ * buffer while the @p manifest is used
+ *
+ * @param[in]   manifest    manifest context to store information in
+ * @param[in]   buf         buffer to parse the manifest from
+ * @param[in]   len         length of the manifest data in the buffer
+ *
+ * @return                  SUIT_OK on parseable manifest
+ * @return                  negative @ref suit_v3_error_t code on error
+ */
+int suit_v3_parse(suit_v3_manifest_t *manifest, const uint8_t *buf, size_t len);
+
+/**
+ * @brief Check a manifest policy
+ *
+ * @param[in]   manifest    manifest context to check the policy for
+ *
+ * @return                  0 on valid manifest policy
+ * @return                  -1 on invalid manifest policy
+ */
+int suit_v3_policy_check(suit_v3_manifest_t *manifest);
+
+/**
+ * @brief Parser a cbor subsequence
+ *
+ * @param[in]   bseq        subsequence value
+ * @param[out]  it          cbor iterator
+ *
+ * @return                  0 on success
+ * @return                  -1 if bseq is not a cbor string
+ * @return                  CborError code on other cbor parser errors
+ */
+int suit_cbor_subparse(nanocbor_value_t *bseq, nanocbor_value_t *it);
```

Comment:
This function is never used, why is it needed here?

Reply:
Good, I was trying to remove this function, apparently I already removed the last user somewhere in the refactoring here.